server/wit.js

```diff
@@ -0,0 +1,78 @@
+const {Wit} = require('node-wit');
+const debug = require('debug')('robat');
+
+function firstEntityValue(entities, entity) {
+  const val = entities && entities[entity] &&
+    Array.isArray(entities[entity]) &&
+    entities[entity].length > 0 &&
+    entities[entity][0].value;
+
+  if (!val) {
+    return null;
+  }
+
+  return typeof val === 'object' ? val.value : val;
+}
+
+module.exports = function (io) {
+  return new Wit({
+    accessToken: process.env.WIT_KEY,
+    actions: {
+      send({sessionId}, {text}) {
+        return new Promise((resolve, reject) => {
+          io.to(sessionId).emit('message', text);
+          return resolve();
+        });
+      },
+      search,
+      displayResults
+    },
+  });
+
+  function search({context, entities}) {
+    return new Promise((resolve, reject) => {
+
+      // Set context based on entities
```

Comment:
Let's make this more modular based on different action files. For every action a different files to set the context based on the entities.